calendar/index.js

```diff
@@ -353,6 +367,7 @@ class IRNMNCalendar extends HTMLElement {
                 dayBtn,
                 `checkin-selected-${this.name}`,
             );
+            this.highlightRange(this.state.checkin, this.state.checkin);
         }
         // If no checkout is set and the selected time is after check-in, set checkout
         else if (!this.state.checkout && time > this.state.checkin) {
```

Comment:
here I recommend switch case instead

Reply:
switch case wouldn't work, sorry. But splitting this into different methods would be better to read